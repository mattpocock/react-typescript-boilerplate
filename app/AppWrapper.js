/**
 * app.js
 *
 * This is the entry file for the application, only setup and boilerplate
 * code.
 */

// Import all the third party stuff
import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import history from 'utils/history';
import 'sanitize.css/sanitize.css';

// Import Language Provider
import LanguageProvider from 'containers/LanguageProvider';

// Load the favicon and the .htaccess file
/* eslint-disable */
import '!file-loader?name=[name].[ext]!./images/favicon.ico';
import 'file-loader?name=.htaccess!./.htaccess';
/* eslint-enable */

import configureStore from './configureStore';
import ThemeProvider from 'ThemeProvider';

// Import i18n messages

// Create redux store with history
const initialState = {};
const store = configureStore(initialState, history);

// eslint-disable-next-line react/prop-types
const AppWrapper = ({ children, messages }) => (
  <ThemeProvider>
    <Provider store={store}>
      <LanguageProvider messages={messages}>
        <ConnectedRouter history={history}>{children}</ConnectedRouter>
      </LanguageProvider>
    </Provider>
  </ThemeProvider>
);

export default AppWrapper;
