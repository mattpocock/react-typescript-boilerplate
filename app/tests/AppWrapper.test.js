import React from 'react';
import { render } from 'react-testing-library';

import AppWrapper from '../AppWrapper';

describe('<AppWrapper />', () => {
  it('should render and match the snapshot', () => {
    const { container } = render(
      <AppWrapper messages={{}}>
        <div />
      </AppWrapper>,
    );
    expect(container).toMatchSnapshot();
  });
});
