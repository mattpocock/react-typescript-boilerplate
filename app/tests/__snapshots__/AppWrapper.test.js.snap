// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppWrapper /> should render and match the snapshot 1`] = `
<div>
  <div />
</div>
`;
